sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1805,9 +1805,9 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
-  val ARITHMETIC_OPERATIONS_FAIL_ON_OVERFLOW =
-    buildConf("spark.sql.arithmeticOperations.failOnOverFlow")
-      .doc("If it is set to true, all arithmetic operations on non-decimal fields throw an " +
+  val FAIL_ON_INTEGER_OVERFLOW =
+    buildConf("spark.sql.failOnIntegerOverFlow")
```

Comment:
failOnIntegerOverFlow -> failOnIntegralTypeOverFlow? To me, Integer is a bit ambiguous.